examples/demo_opts.py

```diff
@@ -9,118 +9,158 @@
-        parser.error(e)
-
-elif args.display in ('pcd8544'):
-    import luma.lcd.device
-    Device = getattr(luma.lcd.device, args.display)
-    try:
-        serial = luma.core.serial.spi(port=args.spi_port,
-                                      device=args.spi_device,
-                                      bus_speed_hz=args.spi_bus_speed,
-                                      bcm_DC=args.bcm_data_command,
-                                      bcm_RST=args.bcm_reset)
-        luma.lcd.device.backlight(bcm_LIGHT=args.bcm_backlight).enable(True)
-        device = Device(serial, rotate=args.rotate)
-
-    except Exception as e:
-        parser.error(e)
-
-elif args.display in ('max7219'):
-    import luma.led_matrix.device
-    Device = getattr(luma.led_matrix.device, args.display)
     try:
-        serial = luma.core.serial.spi(port=args.spi_port,
-                                      device=args.spi_device,
-                                      bus_speed_hz=args.spi_bus_speed,
-                                      gpio=luma.core.serial.noop())
-        device = Device(serial, width=args.width, height=args.height,
-                        rotate=args.rotate, block_orientation=args.block_orientation)
-
-    except Exception as e:
-        parser.error(e)
-
-elif args.display in ('capture', 'pygame', 'gifanim'):
-    import luma.emulator.device
-    Device = getattr(luma.emulator.device, args.display)
-    device = Device(**vars(args))
-
-else:
-    parser.error('unknown display %s' % args.display)
+        import argcomplete
+        argcomplete.autocomplete(parser)
+    except ImportError:
+        pass
+
+    return parser
+
+
+def get_device(actual_args=None):
+    """
+    Create device from command-line arguments and return it.
+    """
+    if actual_args is None:
+        actual_args = sys.argv[1:]
+    parser = create_parser()
+    args = parser.parse_args(actual_args)
+
+    if args.config:
+        # load config from file
+        config = load_config(args.config)
+        args = parser.parse_args(config + actual_args)
+
+    # find display and create device
+    if args.display in display_types.get('oled'):
+        # luma.oled
+        import luma.oled.device
+        Device = getattr(luma.oled.device, args.display)
+        try:
+            if args.interface == 'i2c':
+                from luma.core.serial import i2c
+                serial = i2c(port=args.i2c_port, address=args.i2c_address)
+
+            elif args.interface == 'spi':
+                from luma.core.serial import spi
+                serial = spi(port=args.spi_port,
+                    device=args.spi_device,
+                    bus_speed_hz=args.spi_bus_speed,
+                    bcm_DC=args.bcm_data_command,
+                    bcm_RST=args.bcm_reset)
+
+            device = Device(serial, width=args.width, height=args.height,
+                            rotate=args.rotate, mode=args.mode)
+            return device
+
+        except Exception as e:
```

Comment:
Was wondering about this : should it just trap `luma.core.error` exceptions?
At the moment if there's a bug it gets trapped here, but the traceback is, and I have to go and change the line below to pass, then rerun.

Also, can the `try except` block be moved  out to the top level inside the function rather than being repeated 4 times?